src/map/include/computeMap.hpp

```diff
@@ -541,130 +617,143 @@ namespace skch
-          SlideMapper<Q_Info> slidemap(Q);
-
-          //Initialize iterator over minimizerIndex
-          MIIteratorL2 mi_L2iter( firstSuperWindowRangeStart, firstSuperWindowRangeEnd,
-              countMinimizerWindows);
-
-          //Insert all the minimizers in the first 'super-window'
-          //  [ mi_L2iter.sw_beg, mi_L2iter.sw_end )
-          slidemap.insert_ref(mi_L2iter.sw_beg, mi_L2iter.sw_end);
-
-          auto prev_beg_iter = mi_L2iter.sw_beg;
-          auto prev_end_iter = mi_L2iter.sw_end;
-
-          int beginOptimalPos, lastOptimalPos;
-
-          while ( std::distance(mi_L2iter.sw_end, lastSuperWindowRangeEnd) >= 0)
+#ifdef DEBUG
+          std::cout << "INFO, skch::Map:computeL2MappedRegions, read id " << Q.seqName << "_" << Q.startPos << std::endl; 
+#endif
+          for (L1_candidateLocus_t& l1_candidate : l1_vec_in) 
           {
-            assert( std::distance(mi_L2iter.sw_beg, firstSuperWindowRangeStart) <= 0);
-            assert( std::distance(mi_L2iter.sw_end, lastSuperWindowRangeEnd  ) >= 0);
-
-            //Check if the previous first minimizer is out of current range
-            if (prev_beg_iter != mi_L2iter.sw_beg)
-              slidemap.delete_ref(prev_beg_iter);
-
-            //Check if we have new minimizer in the current range
-            if (prev_end_iter != mi_L2iter.sw_end)
-              slidemap.insert_ref(prev_end_iter);
-          
-            //Is this sliding window the best we have so far?
-            if (slidemap.sharedSketchElements > l2_out.sharedSketchSize)
+//#ifdef DEBUG
+            //std::cout << "Candidate window @ seqId: " << l1_candidate.seqId << ":" << l1_candidate.firstPos << ":" << l1_candidate.lastPos << std::endl;
+//#endif
+           
+            // Get first potential mashimizer
+            const MashimizerInfo first_mashimizer = MashimizerInfo {0, l1_candidate.seqId, l1_candidate.firstPos - param.segLength - 1, 0, 0};
+            //const MashimizerInfo first_mashimizer = MashimizerInfo {0, l1_candidate.seqId, -1, 0, 0};
+            auto firstOpenIt = std::lower_bound(mashimizerIndex.begin(), mashimizerIndex.end(), first_mashimizer); 
+            //std::cout << "First mashimizer @ index " << std::distance(mashimizerIndex.begin(), firstOpenIt) << std::endl;;
+
+            // Keeps track of the lowest end position
+            std::vector<skch::MashimizerInfo> slidingWindow;
+
+            // Used to make a min-heap
+            constexpr auto heap_cmp = [](const skch::MashimizerInfo& l, const skch::MashimizerInfo& r) {return l.wpos_end > r.wpos_end;};
+
+            // windowIt keeps track of beginning of window
+            auto windowIt = firstOpenIt;
+
+            //
+            int windowLen = Q.len - param.segLength;
+            
+            // slideMap tracks the S(A or B) and S(A) and S(B)
+            SlideMapper<Q_Info> slideMap(Q);
+
+            int beginOptimalPos = 0;
+            int lastOptimalPos = 0;
```

Comment:
@bkille, these should variables should be int64_t to prevent overflow when these two variables are averaged (then summed) before being assigned to an int32_t variable. This happens with big plant chromosome sequences.